js/foam/u2/TableView.js

```diff
@@ -0,0 +1,448 @@
+
+  properties: [
+    ['nodeName', 'flex-table'],
+    ['ascIcon', '&#9650;'],
+    ['descIcon', '&#9660;'],
+    {
+      name: 'data',
+      postSet: function(old, nu) {
+        this.model = this.getModel();
+      },
+    },
+    {
+      name: 'model',
+      lazyFactory: function() {
+        return this.getModel();
+      },
+      postSet: function(old, nu) {
+        if (old === nu) return;
+        if (nu && this.allProperties_.length === 0)
+          this.allProperties_ = nu.getRuntimeProperties().filter(
+            function(prop) { return !prop.hidden; });
+      }
+    },
+    {
+      type: 'Array',
+      name: 'allProperties_',
+      documentation: 'All the (non-hidden) properties on the model.',
+    },
+    {
+      type: 'Array',
+      name: 'selectedProperties_',
+      documentation: 'An array of Property objects for all selected ' +
+          'properties. That is, the current set of columns. Defaults to the ' +
+          'model\'s tableProperties if defined, or all non-hidden properties ' +
+          'otherwise.',
+      lazyFactory: function() {
+        if (this.model && this.model.tableProperties &&
+            this.model.tableProperties.length > 0) {
+          return this.model.tableProperties.map(function(name) {
+            return this.model.getProperty(name);
+          }.bind(this));
+        } else {
+          return this.allProperties_;
+        }
+      },
+      postSet: function(old, nu) {
+        this.scrollView.invalidate();
+      },
+    },
+    ['sortOrder', undefined],
+    ['sortProp', undefined],
+    {
+      type: 'Int',
+      name: 'minColWidth',
+      defaultValue: 50
+    },
+    'hardSelection',
+    'softSelection',
+    'headE',
+    'bodyE',
+    {
+      model_: 'foam.core.types.DAOProperty',
+      name: 'filteredDAO',
+      dynamicValue: [function() {
+        this.data; this.sortOrder;
+      }, function() {
+        return this.data.orderBy(this.sortOrder);
+      }]
+    },
+    {
+      type: 'ViewFactory',
+      name: 'rowView',
+      defaultValue: 'foam.u2.TableRowView',
+      documentation: 'Set this to override the row view. It should be a ' +
+          'subclass of $$DOC{ref:"foam.u2.TableRowView"} unless you ' +
+          'really know what you\'re doing.',
+      adapt: function(old, nu) {
+        return nu ? nu : 'foam.u2.TableRowView';
+      }
+    },
+    {
+      name: 'rowHeight',
+      documentation: 'Set this to override the height of a table row.',
+      defaultValue: 48
+    },
+    {
+      name: 'scrollView',
+      factory: function() {
+        return this.ScrollView.create({
+          data: this.filteredDAO$Proxy,
+          rowHeight: this.rowHeight,
+          rowView: this.makeRow,
+        });
+      }
+    },
+    {
+      type: 'Array',
+      name: 'colWidths_',
+      lazyFactory: function() {
```

Comment:
Why is this needed?  ArrayProperty already defines factory.  Either remove or comment.